### ISSUE TITLE
plugin/cache: Add minimum ttl setting

### ISSUE THREAD

**Issue by aaronriekenberg** [coredns/coredns#2054]
Unbound has a cache-min-ttl setting, which is helpful to limit the number of data fetches from the backend.  From https://nlnetlabs.nl/documentation/unbound/unbound.conf/:

```  
              Time  to  live  minimum  for  RRsets  and messages in the cache.
              Default is 0.  If the minimum kicks in, the data is  cached  for
              longer than the domain owner intended, and thus less queries are
              made to look up the data.  Zero makes sure the data in the cache
              is  as the domain owner intended, higher values, especially more
              than an hour or so, can lead to trouble as the data in the cache
              does not match up with the actual data any more.
```

Seems like this would be quite easy to implement here: https://github.com/coredns/coredns/blob/master/plugin/cache/cache.go#L166

Have already gotten TTL from the message and set duration based on this TTL and the max TTL configured.  Just one more check to also clamp duration no lower than min TTL if this new option is set.  Looks like this would affect both the time the entry is stored in the cache and the TTL in the response - which is the same behavior as unbound.

Anyone see a reason not to add an additional min TTL parameter to cache?  Thinking of submitting a PR for this.

**Comment by miekg** [comment]
[ Quoting <notifications@github.com> in "[coredns/coredns] plugin/cache: Add..." ]
>Unbound has a cache-min-ttl setting, which is helpful to limit the number of data fetches from the backend.  From https://nlnetlabs.nl/documentation/unbound/unbound.conf/:
>
>```
>              Time  to  live  minimum  for  RRsets  and messages in the cache.
>              Default is 0.  If the minimum kicks in, the data is  cached  for
>              longer than the domain owner intended, and thus less queries are
>              made to look up the data.  Zero makes sure the data in the cache
>              is  as the domain owner intended, higher values, especially more
>              than an hour or so, can lead to trouble as the data in the cache
>              does not match up with the actual data any more.
>```
>
>Seems like this would be quite easy to implement here: https://github.com/coredns/coredns/blob/master/plugin/cache/cache.go#L166
>
>Have already gotten TTL from the message and set duration based on this TTL and the max TTL configured.  Just one more check to also clamp duration no lower than min TTL if this new option is set.  Looks like this would affect both the time the entry is stored in the cache and the TTL in the response - which is the same behavior as unbound.
>
>Anyone see a reason not to add an additional min TTL parameter to cache?  Thinking of submitting a PR for this.

Yes, this could be usefull. You could add the option to the success and denial 
config line:

`success CAPACITY [TTL]` --> `success CAPACITY [TTL] [MINTTL]`

you have to a bit careful with interaction with the maxttl and the SOA minimum.

/plugin: cache
/label: enhancement

**Comment by miekg** [comment]
Fixed via #2055
